src/components/BlockchainInfo.js

```diff
@@ -592,37 +598,77 @@ class BlockchainInfo extends React.Component {
     }
   }
 
-  async fundAsset(assetId, amount, onSuccessConfirmationPopup, onFailureContributionPopup, amountDollars) {
+  buildAndUpdateNotification(notificationId, type, status, params){
+    this.updateNotification(notificationId, {
+      [type]: {
+        ...params,
+      },
+      status,
+    });
+  }
+
+  fundAsset(assetId, amount, onSuccessConfirmationPopup, onFailureContributionPopup, amountDollars) {
     try {
       const currentAsset = this.state.assets.find(item => item.assetID === assetId);
       const notificationId = Date.now();
+      const {
+        name : assetName,
+      } = currentAsset;
+
+      this.buildAndUpdateNotification(notificationId, NotificationTypes.METAMASK, NotificationStatus.INFO, {
+        operationType: NotificationsMetamask.FUNDING,
+        assetName,
+      });
+
+      const onTransactionHash = () => {
+        this.buildAndUpdateNotification(notificationId, NotificationTypes.FUNDING, NotificationStatus.INFO, {
+          assetName,
+          amount: amountDollars,
+        });
+      }
+
+      const onReceipt = (wasSuccessful) => {
+        if(wasSuccessful){
+          onSuccessRefreshData();
+        } else {
+          onFailureContributionPopup();
+          this.buildAndUpdateNotification(notificationId, NotificationTypes.FUNDING, NotificationStatus.ERROR, {
+            assetName,
+          });
+        }
+      }
+
+      const onError = (type) => {
+        onFailureContributionPopup();
+        if(type === 'metamask'){
```

Comment:
Maybe we can have something like`ErrorTypes.METAMASK` and so on.

Reply:
I knew you were gonna say something about that ahah I'll add it

Reply:
You know me by now ❤️